jose/validate.go

```diff
@@ -49,8 +50,12 @@ func validateX5(certs []*x509.Certificate, key interface{}) error {
 		return errors.New("certs cannot be empty")
 	}
 
-	if err := keyutil.VerifyPair(certs[0].PublicKey, key); err != nil {
-		return errors.Wrap(err, "error verifying certificate and key")
+	// We cant get the private key data to directly compare, so don't actually check the keys match if we have
+	// an OpaqueSigner. Perhaps the public key can be derived from the OpaqueSigner and matched?
+	if _, isOpaqueSigner := key.(jose.OpaqueSigner); !isOpaqueSigner {
```

Comment:
We can use our alias:

```suggestion
	if _, isOpaqueSigner := key.(OpaqueSigner); !isOpaqueSigner {
```

Reply:
We should match the public key. `*rsa.PublicKey`, `*ecdsa.PublicKey` and `ed25519.PublicKey` implements `Equal(x crypto.PublicKey) bool`